drivers/storage/portworx/portworx.go

```diff
@@ -652,6 +654,42 @@ func defaultPortworxImageVersion(releases *manifest.ReleaseManifest) string {
 	return defaultPortworxVersion
 }
 
+func setNodeSpecDefaults(toUpdate *corev1alpha1.StorageCluster) {
+	if len(toUpdate.Spec.Nodes) == 0 {
+		return
+	}
+
+	updatedNodeSpecs := make([]corev1alpha1.NodeSpec, 0)
+	for _, nodeSpec := range toUpdate.Spec.Nodes {
+		nodeSpecCopy := nodeSpec.DeepCopy()
+		if nodeSpec.Storage == nil {
+			nodeSpecCopy.Storage = toUpdate.Spec.Storage.DeepCopy()
```

Comment:
check here if toUpdate.Spec.Storage != nil

Reply:
Or this could be refactored to
```
if toUpdate.Spec.Storage != nil {
     		if nodeSpec.Storage == nil {
			nodeSpecCopy.Storage = toUpdate.Spec.Storage.DeepCopy()
                } else {
                          ........
               }
}
```

Reply:
The DeepCopy() handles a nil receiver. It returns nil which is what we want.